src/gsy_e/models/market/balancing.py

```diff
@@ -159,11 +152,10 @@ def _determine_offer_price(
             energy, trade_rate, energy_portion, orig_offer_price)
 
     def accept_offer(  # pylint: disable=too-many-locals
-            self, offer_or_id: Union[str, BalancingOffer], buyer: str, *,
+            self, offer_or_id: Union[str, BalancingOffer], buyer: TraderDetails, *,
             energy: int = None,
             already_tracked: bool = False, trade_rate: float = None,
-            trade_bid_info: float = None,
-            buyer_origin=None, buyer_origin_id=None, buyer_id=None) -> BalancingTrade:
+            trade_bid_info: float = None) -> BalancingTrade:
```

Comment:
Yeah, I guess the reason why this both got away without us noticing and that you cannot find any usage for it is because this is the balancing market. Let me check if I can completely remove it without too much hassle.